clients/src/main/java/org/apache/kafka/common/record/DefaultRecord.java

```diff
@@ -431,7 +431,7 @@ private static void skipBytes(InputStream in, int bytesToSkip) throws IOExceptio
 
         // Starting JDK 12, this implementation could be replaced by InputStream#skipNBytes
         while (bytesToSkip > 0) {
-            long ns = in.skip(bytesToSkip);
+            int ns = (int) in.skip(bytesToSkip);
```

Comment:
New issue introduced in trunk after this PR was originally created. There are a few others, but I'll address them together with the JDK 20 CI build (which will prevent the issue altogether).